src/main/java/emissary/core/IBaseDataObjectHelper.java

```diff
@@ -37,15 +37,27 @@ private InternalIdBaseDataObject(final UUID internalId) {
 
     private IBaseDataObjectHelper() {}
 
+    /**
+     * Clones an IBaseDataObject.
+     * 
+     * @param iBaseDataObject the IBaseDataObject to be cloned.
+     * @return the clone of the IBaseDataObject passed in.
+     */
+    public static IBaseDataObject clone(final IBaseDataObject iBaseDataObject) {
+        return clone(iBaseDataObject, true);
```

Comment:
I'm not a fan having the new overload reference the very method we're deprecating.  Maybe we should refactor the existing one by extracting two supporting methods that both overloads can use.  Something like the following signatures:
```java
private static copyBasicFields(final IBaseDataObject src, final IBaseDataObject dest) {
    // body would be current lines 66-87 
}
private static copyNonBasicFields(final IBaseDataObject src, final IBaseDataObject dest) {
    // body would be current lines 90-106
}
```
With that change, this overload would strip down to the following:
```java
 public static IBaseDataObject clone(final IBaseDataObject src) {
    Validate.notNull(src, "Required: src not null");
    IBaseDataObject dst = new InternalIdBaseDataObject(src.getInternalId());
    copyBasicFields(src, dst);
    copyNonBasicFields(src, dst);
    retrun dst;
}
```
The original now-deprecated overload would become this:
```java
 public static IBaseDataObject clone(final IBaseDataObject src, final boolean fullClone) {
    Validate.notNull(src, "Required: src not null");
    IBaseDataObject dst = fullClone ? new InternalIdBaseDataObject(src.getInternalId()) : DataObjectFactory.getInstance();
    copyBasicFields(src, dst);
    if (fullClone) {
        copyNonBasicFields(src, dst);
    }
    retrun dst;
}
```

After the deprecated overload is removed, we can collapse those private methods into the remaining `clone` method